File: app/src/main/java/com/projecthub/repository/CohortRepository.java
package com.projecthub.repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import org.springframework.stereotype.Repository;

import com.projecthub.model.Cohort;

/**
 * Repository interface for {@link Cohort} entities.
 */
@Repository

public interface CohortRepository {

    Cohort save(Cohort cohort);

    Optional<Cohort> findById(UUID id);

    List<Cohort> findAll();

    void delete(Cohort cohort);

    List<Cohort> findBySchoolId(UUID schoolId);

    Cohort getReferenceById(UUID cohortId);

    void deleteById(UUID id);
}

File: src/main/kotlin/com/projecthub/domain/cohort/CohortRepository.kt
package com.projecthub.domain.cohort

/**
 * Repository interface for cohorts (outbound port)
 * This interface defines the persistence operations available for Cohort entities
 */
interface CohortRepository {
    fun findById(id: String): Cohort?
    fun findAll(): List<Cohort>
    fun findBySchool(schoolId: String): List<Cohort>
    fun save(cohort: Cohort): Cohort
    fun deleteById(id: String)
    fun findActiveCohortsWithCapacity(minCapacity: Int): List<Cohort>
}